crates/pathfinder/src/state/sync.rs

```diff
@@ -419,6 +426,24 @@ where
                         }
                     }
                 }
+                Some(l2::Event::QueryHead(tx)) => {
+                    // TODO add some timeout on the p2p head
```

Comment:
I was thinking we could use a timeout here to fall back to the sequencer checking the L2 head as otherwise we could be stuck at some low head if the new block events stall for some reason.